### PR TITLE
[Doop] Add mock data and remove useEndlessScroll hook

### PR DESCRIPTION
**Pull request by hodanoori**
plus:
1.  implementing mock data handling using our utils lib
2. debounce setSearchterm to avoid unnecessary re-renders

### CODE REVIEW

File: doop/ui/src/components/AppContent.jsx [hunk 1]
```diff
@@ -16,10 +16,39 @@ import { fetchData } from "../lib/apiClient"
...
+        ...{ mock: true },
+      })
+        .then((response) => {
+          if (!response.ok) {
+            addMessage({
+              variant: "error",
+              text: parseError(e.message),
+            })
+          }
+          return response.json()
```
Comment (andypf) [inline doop/ui/src/components/AppContent.jsx:45]: It means that an attempt will be made to convert the response to JSON in any case, even if response.ok is false?

File: doop/ui/src/components/filters/FilterSelect.jsx [hunk 2]
```diff
@@ -87,9 +97,7 @@ const FilterSelect = () => {
       <SearchInput
         className="w-96"
         value={searchValue || ""}
-        onChange={(e) => {
-          setSearchTerm(e.target.value)
-        }}
+        onChange={(value) => handleSearchChange(value)}
```
Comment (andypf) [inline doop/ui/src/components/filters/FilterSelect.jsx:100]: 👍

File: doop/ui/src/lib/apiClient.js [hunk 3]
```diff
@@ -5,13 +5,6 @@
 
 // Example fetch call. Adjust as needed for your API
 export const fetchData = ({ queryKey, meta }) => {
-  if (meta.mock === "true" || meta.mock === true) {
```
Comment (andypf) [inline doop/ui/src/lib/apiClient.js:8]: Perhaps the code was well placed here?

File: doop/ui/db.json [hunk 4]
```diff
@@ -0,0 +1,452 @@
+{
```
Comment (andypf) [inline doop/ui/db.json:1]: 👍

File: doop/ui/src/components/AppContent.jsx (hunk 1 above)
Comment (hodanoori) [reply doop/ui/src/components/AppContent.jsx:45]: I tested this, return response.json() is needed in order to catch the result in the next then, as this promise is pending here and not resolved yet.

File: doop/ui/src/components/AppContent.jsx [hunk 5]
```diff
@@ -16,10 +16,39 @@ import { fetchData } from "../lib/apiClient"
...
+        })
+        .then((result) => {
+          setData(result)
+        })
+    }
+  }, [isMock])
 
   // LOAD DATA
   const dataRequest = useQuery({
```
[the remark this answers is not in the document]
Comment (hodanoori) [reply doop/ui/src/components/AppContent.jsx:54]: I handle this adding enabled:!isMock prop here.

**Review by andypf — APPROVED** [review]

**Pull request merged**